src/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -56,54 +60,93 @@ makeRawPage
     -> RawPage
 makeRawPage ba off = RawPage (div2 off) ba
 
+rawPageRawBytes :: RawPage -> RawBytes
+rawPageRawBytes (RawPage off ba) =
+    RawBytes (P.Vector (mul2 off) 4096 ba)
+
 -------------------------------------------------------------------------------
 -- Lookup function
 -------------------------------------------------------------------------------
 
-rawPageEntry
+data RawPageLookup entry =
+       -- | The key is not present on the page.
+       LookupEntryNotPresent
+
+       -- | The key is present and corresponds to a normal entry that fits
+       -- fully within the page.
+     | LookupEntryNormal !entry
```

Comment:
`Normal` is an overloaded term, because we also use it for "normal" table handles. Maybe just `LookupEntry`?